### PR TITLE
[translation] add continuation token tests

### PR DESCRIPTION
**Pull request by kristapratico**
Forgot to add tests for this scenario

### CODE REVIEW

File: sdk/translation/azure-ai-translation-document/tests/test_translation.py [hunk 1]
```diff
@@ -423,3 +423,18 @@ def test_overloaded_bad_input(self, client):
...
+        source_container_sas_url = self.create_source_container(data=Document(data=b'hello world'))
+        target_container_sas_url = self.create_target_container()
+
+        initial_poller = client.begin_translation(source_container_sas_url, target_container_sas_url, "es")
+        cont_token = initial_poller.continuation_token()
+
+        poller = client.begin_translation(None, continuation_token=cont_token)
+        result = poller.result()
+        assert result
+        initial_poller.wait()  # necessary so azure-devtools doesn't throw assertion error
```
Comment (maririos) [inline sdk/translation/azure-ai-translation-document/tests/test_translation.py:440]: ohh interesting, is there a rule or something that if there is a poller it needs to be awaited?

File: sdk/translation/azure-ai-translation-document/tests/test_translation.py [hunk 2]
```diff
@@ -423,3 +423,18 @@ def test_overloaded_bad_input(self, client):
...
+    @DocumentTranslationClientPreparer()
+    def test_translation_continuation_token(self, client):
+        source_container_sas_url = self.create_source_container(data=Document(data=b'hello world'))
+        target_container_sas_url = self.create_target_container()
+
+        initial_poller = client.begin_translation(source_container_sas_url, target_container_sas_url, "es")
+        cont_token = initial_poller.continuation_token()
+
+        poller = client.begin_translation(None, continuation_token=cont_token)
+        result = poller.result()
```
Comment (maririos) [inline sdk/translation/azure-ai-translation-document/tests/test_translation.py:438]: how do you know the poller is done and u can access the result here?

File: sdk/translation/azure-ai-translation-document/tests/test_translation.py (hunk 1 above)
Comment (kristapratico) [reply sdk/translation/azure-ai-translation-document/tests/test_translation.py:440]: our test framework yells that all LROs need to be resulted or waited if you don't include that

File: sdk/translation/azure-ai-translation-document/tests/test_translation.py (hunk 2 above)
Comment (kristapratico) [reply sdk/translation/azure-ai-translation-document/tests/test_translation.py:438]: Sure, I can add validation for that

**Review by maririos — APPROVED** [review]

**Pull request merged**